src/ray/core_worker/transport/actor_task_submitter.cc

```diff
@@ -40,6 +40,75 @@ void ActorTaskSubmitter::AddActorQueueIfNotExists(const ActorID &actor_id,
           actor_id, execute_out_of_order, max_pending_calls, fail_if_actor_unreachable));
 }
 
+Status ActorTaskSubmitter::SubmitActorCreationTask(TaskSpecification task_spec) {
+  RAY_CHECK(task_spec.IsActorCreationTask());
+  RAY_LOG(DEBUG).WithField(task_spec.TaskId()) << "Submitting actor creation task";
+  resolver_.ResolveDependencies(task_spec, [this, task_spec](Status status) mutable {
+    // NOTE: task_spec here is capture copied (from a stack variable) and also
+    // mutable. (Mutations to the variable are expected to be shared inside and
+    // outside of this closure).
```

Comment:
I think it's sharing the mutations because the underlying is a shared_ptr. If it's a regular class it's copied and outside of the closure we won't see it.